File: app/src/main/java/com/spacemonster/webdataviewer/content/parser/JsoupWebImageParser.java
package com.spacemonster.webdataviewer.content.parser;

import com.spacemonster.webdataviewer.content.data.WebImage;

/**
 * Created by GE62 on 2017-12-26.
 */

public class JsoupWebImageParser implements IDataParser<String, WebImage> {

    @Override
    public WebImage parse(String data) {
        return new WebImage(data);
    }

    @Override
    public void release() {
    }
}

File: app/src/main/kotlin/com/spacemonster/webdataviewer/content/parser/JsoupWebImageParser.kt
package com.spacemonster.webdataviewer.content.parser

import com.spacemonster.webdataviewer.content.data.WebImage
import io.reactivex.Observable

/**
 * Created by GE62 on 2017-12-26.
 */

class JsoupWebImageParser : IDataParser<String, WebImage> {
    override fun parse(dataList: Observable<String>): Observable<WebImage> {
        return dataList.map { WebImage(it) }
    }

    override fun release() {}
}
